docs/disk_cleaning.md

```diff
@@ -0,0 +1,140 @@
+# Ironic automated cleaning in Metal3
+
+This document explains how to disable or enable Ironic's automated cleaning
+feature from Cluster-api-provider-metal3 (CAPM3) Custom Resources (CR).
+
+## Table of Contents
+
+- [Ironic automated cleaning in Metal3](#ironic-automated-cleaning-in-metal3)
+  - [Table of Contents](#table-of-contents)
+  - [What is automated cleaning](#what-is-automated-cleaning)
+  - [How to enable/disable automated cleaning via Metal3](#how-to-enabledisabled-automated-cleaning-via-metal3)
+    - [Via ironic.conf](#via-ironicconf)
+    - [Via CAPM3 APIs](#via-capm3-apis)
+      - [Which resources do I need to edit](#which-resources-do-i-need-to-edit)
+      - [When should I edit my CRs to disable cleaning](#when-should-i-edit-my-crs-to-disable-cleaning)
+      - [Selective configuration](#selective-configuration)
+
+How we refer to these objects later on:
+
+- Metal3Machine - machine
+- Metal3MachineTemplate - template
+- BareMetalHost - host
+
+## What is automated cleaning
+
+Ironic offers two modes for node cleaning, automated and manual where automated
+cleaning is recommended for ironic deployments. Automated cleaning gets triggered
+automatically during the first provisioning and on every deprovisioning of a node.
+If automated cleaning is enabled, Ironic runs various cleaning steps by priority
+and meanwhile the node stays in `cleaning` state (machine/host in
+`deprovisioning`). To get more info, check Ironic's [documentation](https://docs.openstack.org/ironic/latest/admin/cleaning.html#automated-cleaning).
```

Comment:
If "Automated cleaning gets triggered automatically during the first provisioning" then is this line correct "(machine/host in `deprovisioning`)" , shouln't it also be provisioning ?

Reply:
Cleaning takes place on every deprovisioning (active -> available) or first registration (manageable -> available).  As such, I don't think BMH can be in `provisioning` state when Ironic node is in `cleaning` state.
Reference to the Ironic host state machine:  https://docs.openstack.org/ironic/latest/admin/cleaning.html#automated-cleaning

Reply:
but my point is can it only be in deprovisioning state?

Reply:
not only, but I don't know to be honest if it is preparing state or inspecting state when it is first time

Reply:
Let's remove that part not to confuse ourself + readers